src/ReleaseReport/ReleaseReportGenerator.cs

```diff
@@ -0,0 +1,115 @@
+    {
+        string version = matrix?.ChannelVersion
+                         ?? majorRelease?.ChannelVersion
+                         ?? "0.0";
+        HttpClient client = new();
+        DateOnly threeMonthsDate = DateOnly.FromDateTime(DateTime.UtcNow.AddMonths(3));
+        DateOnly initialRelease = majorRelease?.Releases.FirstOrDefault(r => r.ReleaseVersion.Equals($"{version}.0"))?.ReleaseDate ?? DateOnly.MaxValue;
+        DateOnly eolDate = majorRelease?.EolDate ?? DateOnly.MaxValue;
+        ReportOverview report = new(DateTime.UtcNow, version, []);
+
+        foreach (SupportFamily family in matrix?.Families ?? throw new Exception())
+        {
+            ReportFamily reportFamily = new(family.Name, []);
+            report.Families.Add(reportFamily);
+
+            foreach (SupportDistribution distro in family.Distributions)
+            {
+                IList<SupportCycle>? cycles;
+                List<string> activeReleases = [];
+                List<string> unsupportedActiveRelease = [];
+                List<string> soonEolReleases = [];
+                List<string> supportedUnActiveReleases = [];
+                List<string> unsupportedUnActiveReleases = [];
+                List<string> missingReleases = [];
+                
+                try
+                {
+                    cycles = await EndOfLifeDate.Product.GetProduct(client, distro.Id);
+                    if (cycles is null)
+                    {
+                        continue;
+                    }
+                }
+                catch (HttpRequestException)
+                {
+                    Console.WriteLine($"No data found at endoflife.date for: {distro.Id}");
+                    Console.WriteLine();
+                    continue;
+                }
+
+                foreach (SupportCycle cycle in cycles)
+                {
+                    SupportInfo support = cycle.GetSupportInfo();
+                    // dotnet statements
+                    bool isSupported = distro.SupportedVersions.Contains(cycle.Cycle);
+                    bool isUnsupported = distro.UnsupportedVersions?.Contains(cycle.Cycle) ?? false;
+                    // EndofLife.Date statement
+                    bool isActive = support.IsActive;
+                    bool hasOverlappingLifecycle = initialRelease <= support.EolDate && cycle.ReleaseDate <= eolDate;
+
+                    if (isActive)
+                    {
+                        activeReleases.Add(cycle.Cycle);
+                    }
+
+                    /*
+                        Cases (EOLDate, dotnet):
+                        1. (Active, Supported)
+                        2. (Active, Unsupported)
+                        3. (OverlappingLifecycle, Unlisted)
+                        4. (EOL, Supported)
+                        5. (Active - EolSoon, Supported)
+                        6. (EOL, Unsupported | Unlisted)
```

Comment:
Having that case would allow me to check for former supported versions and adding them to a table. That's the only thing that was blocking me from using this report directly and needing to duplicate the whole logic.